packages/@aws-cdk/toolkit-lib/test/toolkit-flags.test.ts

```diff
@@ -0,0 +1,251 @@
+import { ArtifactType } from '@aws-cdk/cloud-assembly-schema';
+import type { CloudAssembly } from '@aws-cdk/cx-api';
+import { appFixture, TestIoHost } from './_helpers';
+import { Toolkit } from '../lib/toolkit/toolkit';
+let ioHost: TestIoHost;
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+  ioHost = new TestIoHost();
+});
+
+function createMockCloudAssemblySource(artifacts: any) {
+  return {
+    async produce() {
+      const mockCloudAssembly = {
+        manifest: {
+          artifacts: artifacts,
+        },
+      } as CloudAssembly;
+
+      return {
+        cloudAssembly: mockCloudAssembly,
+        dispose: jest.fn(),
+        [Symbol.asyncDispose]: jest.fn(),
+        _unlock: jest.fn(),
+      };
+    },
+  };
+}
+
+describe('Toolkit.flags() method', () => {
+  test('should retrieve feature flags in correct structure', async () => {
+    const toolkit = new Toolkit({ ioHost });
+    const cx = await appFixture(toolkit, 'two-empty-stacks');
+    const flags = await toolkit.flags(cx);
+
+    expect(flags.length).toBeGreaterThan(0);
+    expect(Array.isArray(flags)).toBe(true);
+
+    const expectedFlags = [
+      '@aws-cdk/aws-s3:createDefaultLoggingPolicy',
+      '@aws-cdk/core:newStyleStackSynthesis',
+      '@aws-cdk/aws-apigateway:usagePlanKeyOrderInsensitiveId',
+    ];
+
+    let foundFlags = 0;
+
+    flags.forEach((report) => {
+      expect(Object.keys(report.flags).length).toBeGreaterThan(0);
+      expect(report).toHaveProperty('module');
+      expect(report).toHaveProperty('flags');
+      expect(typeof report.module).toBe('string');
```

Comment:
This implies the `toHaveProperty`, so if you do this you don't need the other one. But why are we doing this?